### PR TITLE
3037 rops rbac groups admin page

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context
add Group Member admin   to ROPS

# What has changed
Add new screens/code to achieve this

# How to test?
Setup some groups, members and admins in support tool.  Then view and try to remove them in this version of ROPS

# Links

# Screenshots (if appropriate):
](https://trello.com/b/yarvSxGu/generic-response-management)

### CODE REVIEW

File: ui/src/DeleteUserFromGroupConfirmation.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      </Helmet>
+      {errorSummary.length > 0 && <ErrorSummary />}
+      <p>
+        Are you sure you wish to remove {props.userEmail} user? from{" "}
+        {props.groupName}
+      </p>
+      <button type="button" className="btn" onClick={removeUser}>
+        <span className="btn__inner">Yes</span>
+      </button>
+      {/* Todo How to do a cancel button? */}
```
Comment (NickGrantONS) [inline ui/src/DeleteUserFromGroupConfirmation.js:90]: Why is this TODO left here?

File: ui/src/GroupAdmin.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      const allGroupMembersResponse = await fetch(
+        `/api/userGroupMembers/findByGroup/${props.groupId}`
+      );
+      const allUsersInGroup = await allGroupMembersResponse.json();
+
+      const usersInGroupRows = await allUsersInGroup.map((groupUser, index) => (
+        <tr className="table__row" key={index}>
+          <td className="table__cell">{groupUser.userEmail}</td>
+          <td className="table__cell">
+            <button onClick={() => openRemoveUserPage(groupUser)} type="button">
```
Comment (NickGrantONS) [inline ui/src/GroupAdmin.js:28]: this button does not follow any government standards (GDS) or the ONS pattern library... it's just a completely unstyled plain HTML button.

File: ui/src/GroupAdmin.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    fetchAllUsersInGroup();
+  }, [props.groupId]);
+
+  return (
+    <>
+      <Helmet>
+        <title>View Group</title>
+      </Helmet>
+      <Link to="/">← Back to home</Link>
```
Comment (NickGrantONS) [inline ui/src/GroupAdmin.js:45]: Should go back to "my groups" not to home

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
Problem summary:

1. Need to use government(GDS)/ONS UI standards... pattern library... can't just make things up or leave things unstyled
2. Every page needs a title, can't just have tables
3. Group details page should route back to the page it came from (i.e. "my groups" not home)
4. Table headings are for table headings. If you need to put other stuff on the screen those need to be their own UI elements... don't put everything into table headings because that's not accessible or good UI design
5. Only use the announcer to announce banners, not everything on the screen
6. See other comments

Screenshot 2021-11-24 at 13 15 29
Buttons not styled. Table header row has got other stuff in it which is nothing to do with table heading. No page heading. Back button should go back to "my groups".

Screenshot 2021-11-24 at 13 14 37
Table header row has got other stuff in it which is nothing to do with table heading. No page heading.

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
The features seem to work fine but a few styling/wording points:
- Should have consistent capitalization on headings e.g. "Group Member email"
- The "remove" buttons on the user appear to be using the "call to action" style where they should probably just be plain buttons
- The yes and cancel buttons on the confirmation screen should use the grouped buttons styling so that the cancel button has the "secondary negative action" styling 
- Broken wording on the confirm user removal page e.g.  "Are you sure you wish to remove dummy@example.com user? from TEST GROUP"
- Slightly janky styling on the group table with the remove buttons floating out to the right of the table heading line

**Review by AdamHawtin — APPROVED** [review]
All seems to work as expected. 👍 

I'm not sure about the styling/placement of the "Back to ..."  buttons, especially when combined with a flashed message, but I think that's something to come back to with a site wide pattern for, especially since many existing pages are currently missing any form of back button entirely.

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
Need to add description to table of groups and I think you should add it to the group user admin screen too. It will require API changes.

Or, we could say that because group description was added since originally tasking & sizing this ticket, it's beyond the scope... but it would be great to get it done.

Also, the "remove" button does look like a call to action, as Adam said... should be a secondary button, no?

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
Screenshot 2021-11-26 at 17 37 25

Table headings don't line up with table rows

Screenshot 2021-11-26 at 17 38 14

Remove button is still a massive "call to action" button, not a small grey secondary button

**Review by NickGrantONS — APPROVED** [review]
My comments have all been addressed. Great work. Looks good 👍

**Pull request merged**